components/bl-particles-component/src/helpers.js

```diff
@@ -0,0 +1,191 @@
+const RotateDegrees = { MIN: 0, MAX: 360 };
+const SpiralShapeProps = {
+  particles  : { stroke: { width: STROKE_WIDTH } },
+  innerRadius: 1,
+  lineSpacing: 1,
+  fill       : false,
+  close      : false,
+};
+
+const LineShapeProps = {
+  particles: {
+    stroke: { width: STROKE_WIDTH },
+    move  : { straight: true },
+    rotate: { path: true },
+  },
+};
+
+export function useOptions(component) {
+  const {
+    preset, textValue, color, backgroundColor, autoPlay, fullScreen, shape, zIndex, lineLinksVisibility, move, size,
+    speed, opacity, direction, colorAnimation, number, imageURL, rotate, rotationAnimation, customOptions, linksColor,
+    rollingAnimation, duration, delay, triangleLinksVisibility, collisionMode, outModes, backgroundOpacity,
+  } = component;
+
+  const particles = useMemo(() => ({
+    collisions: { enable: collisionMode !== UNSET_COLLISION_MODE, mode: collisionMode },
+    links     : {
+      enable   : lineLinksVisibility || triangleLinksVisibility,
+      distance : LINKS_DISTANCE,
+      color    : linksColor || Colors.RANDOM,
+      opacity  : OPACITY[lineLinksVisibility ? 'FULL' : 'LOW'],
+      triangles: {
+        enable : triangleLinksVisibility,
+        color  : linksColor,
+        opacity: OPACITY.LOW,
+      },
+    },
+    rotate    : {
+      value    : RotateDegrees[rotate ? 'MAX' : 'MIN'],
+      random   : true,
+      direction: Direction.RANDOM,
+      animation: { enable: rotationAnimation, speed, sync: false },
+    },
+    roll      : { enable: rollingAnimation, speed },
+    color     : {
+      value    : color || Colors.RANDOM,
+      animation: { enable: colorAnimation, speed: COLOR_ANIMATION_SPEED },
+    },
+    opacity   : { value: opacity },
+    shape     : {
+      type   : shape,
+      options: {
+        [Shapes.IMAGE] : Array.isArray(imageURL) ? imageURL : { src: imageURL },
+        [Shapes.TEXT]  : { value: textValue },
+        [Shapes.SPIRAL]: SpiralShapeProps,
+        [Shapes.LINE]  : LineShapeProps,
+      },
+    },
+    move      : { enable: move, direction, speed, outModes },
+    size      : { value: size },
+    number    : { value: number, density: { enable: true } },
+  }), [
+    collisionMode, lineLinksVisibility, triangleLinksVisibility, linksColor, rotate, rotationAnimation, speed, number,
+    rollingAnimation, color, colorAnimation, opacity, shape, imageURL, textValue, move, direction, outModes, size,
+  ]);
+
+  return useMemo(() => ({
+    autoPlay, particles,
+    duration    : duration || DEFAULT_TIMING,
+    delay       : delay || DEFAULT_TIMING,
+    fullScreen  : { enable: fullScreen, zIndex: zIndex || DEFAULT_Z_INDEX },
+    background  : { color: backgroundColor, opacity: backgroundOpacity },
+    detectRetina: true,
+    pauseOnBlur : false,
+    ...PresetsMap[preset],
+    ...customOptions,
+  }), [
+    autoPlay, particles, duration, delay, fullScreen, zIndex, backgroundColor, backgroundOpacity, preset, customOptions,
+  ]);
```

Comment:
This useMemo shouldn't depend on `particles`, `autoPlay` and `customOptions` as i see

```suggestion
  const otherProps = useMemo(() => ({
    duration    : duration || DEFAULT_TIMING,
    delay       : delay || DEFAULT_TIMING,
    fullScreen  : { enable: fullScreen, zIndex: zIndex || DEFAULT_Z_INDEX },
    background  : { color: backgroundColor, opacity: backgroundOpacity },
    ...PresetsMap[preset],
  }), [ duration, delay, fullScreen, zIndex, backgroundColor, backgroundOpacity, preset  ]);
  
  
  return {
    particles, autoPlay,
     ...otherProps,    ...customOptions, 
     detectRetina: true,    pauseOnBlur : false 
   }
```